### PR TITLE
Add integrations tests to check malformed API contexts

### PR DESCRIPTION
**Pull request by dulithsenanayake**
This PR adds integration tests to cover the following scenarios.

- Create publisher rest API with malformed context.
- Create graphql API using schema with malformed context.
- Create API product with malformed context.
- Create websocket API with malformed context.
- WSDL import with malformed context.

### CODE REVIEW

**Comment by BLasan** [comment]
Build is failed

**Review by BLasan — DISMISSED** [review]

**Comment by BLasan** [comment]
```
2023-07-09T20:07:49.5566524Z [ERROR] Tests run: 1499, Failures: 11, Errors: 0, Skipped: 46, Time elapsed: 17,640.501 s <<< FAILURE! - in TestSuite
2023-07-09T20:07:49.5570125Z [ERROR] org.wso2.am.integration.tests.apiproduct.APIProductCreationTestCase.cleanUpArtifacts  Time elapsed: 0.772 s  <<< FAILURE!
2023-07-09T20:07:49.5571092Z org.wso2.am.integration.clients.publisher.api.ApiException: 
2023-07-09T20:07:49.5572269Z 	at org.wso2.am.integration.tests.apiproduct.APIProductCreationTestCase.cleanUpArtifacts(APIProductCreationTestCase.java:644)
2023-07-09T20:07:49.5573998Z 
2023-07-09T20:07:49.5575542Z [ERROR] org.wso2.am.integration.tests.apiproduct.APIProductCreationTestCase.cleanUpArtifacts  Time elapsed: 1.525 s  <<< FAILURE!
2023-07-09T20:07:49.5576898Z org.wso2.am.integration.clients.publisher.api.ApiException: Missing the required parameter 'apiId' when calling deleteAPI(Async)
2023-07-09T20:07:49.5577914Z 	at org.wso2.am.integration.tests.apiproduct.APIProductCreationTestCase.cleanUpArtifacts(APIProductCreationTestCase.java:644)
2023-07-09T20:07:49.5579323Z 
2023-07-09T20:07:49.5581025Z [ERROR] org.wso2.am.integration.tests.api.lifecycle.APISecurityTestCase.initialize  Time elapsed: 4.057 s  <<< FAILURE!
2023-07-09T20:07:49.5581807Z org.wso2.am.integration.clients.publisher.api.ApiException: 
2023-07-09T20:07:49.5582584Z 	at org.wso2.am.integration.tests.api.lifecycle.APISecurityTestCase.initialize(APISecurityTestCase.java:190)
2023-07-09T20:07:49.5583021Z 
2023-07-09T20:07:49.5585888Z [ERROR] org.wso2.am.integration.tests.api.lifecycle.APISecurityTestCase.cleanUpArtifacts  Time elapsed: 4.157 s  <<< FAILURE!
2023-07-09T20:07:49.5587196Z org.wso2.am.integration.clients.publisher.api.ApiException: Missing the required parameter 'apiId' when calling deleteAPI(Async)
2023-07-09T20:07:49.5587948Z 	at org.wso2.am.integration.tests.api.lifecycle.APISecurityTestCase.cleanUpArtifacts(APISecurityTestCase.java:1206)
2023-07-09T20:07:49.5589658Z 
2023-07-09T20:07:49.5591258Z [ERROR] org.wso2.am.integration.tests.other.WSDLImportTestCase.testWsdlDefinitionImportWithMalformedContext  Time elapsed: 0.113 s  <<< FAILURE!
2023-07-09T20:07:49.5592081Z java.lang.AssertionError: expected [400] but found [409]
2023-07-09T20:07:49.5593012Z 	at org.wso2.am.integration.tests.other.WSDLImportTestCase.testWsdlDefinitionImportWithMalformedContext(WSDLImportTestCase.java:292)
2023-07-09T20:07:49.5593464Z 
2023-07-09T20:07:49.5597023Z [ERROR] org.wso2.am.integration.tests.publisher.APIM18CreateAnAPIThroughThePublisherRestAPITestCase.testCreateAnAPIWithMalformedContextThroughThePublisherRest  Time elapsed: 0.231 s  <<< FAILURE!
2023-07-09T20:07:49.5598282Z java.lang.AssertionError: Response cannot be null expected object to not be null
2023-07-09T20:07:49.5599419Z 	at org.wso2.am.integration.tests.publisher.APIM18CreateAnAPIThroughThePublisherRestAPITestCase.testCreateAnAPIWithMalformedContextThroughThePublisherRest(APIM18CreateAnAPIThroughThePublisherRestAPITestCase.java:172)
2023-07-09T20:07:49.5601173Z 
2023-07-09T20:07:49.5603526Z [ERROR] org.wso2.am.integration.tests.graphql.GraphqlTestCase.testCreateAndPublishGraphQLAPIUsingSchemaWithMalformedContext  Time elapsed: 0.447 s  <<< FAILURE!
2023-07-09T20:07:49.5604344Z java.lang.AssertionError: Response cannot be null expected object to not be null
2023-07-09T20:07:49.5605122Z 	at org.wso2.am.integration.tests.graphql.GraphqlTestCase.testCreateAndPublishGraphQLAPIUsingSchemaWithMalformedContext(GraphqlTestCase.java:226)
2023-07-09T20:07:49.5606934Z 
2023-07-09T20:07:49.5609109Z [ERROR] org.wso2.am.integration.tests.graphql.GraphqlTestCase.testCreateAndPublishGraphQLAPIUsingSchemaWithMalformedContext  Time elapsed: 0.242 s  <<< FAILURE!
2023-07-09T20:07:49.5609879Z java.lang.AssertionError: Response cannot be null expected object to not be null
2023-07-09T20:07:49.5610645Z 	at org.wso2.am.integration.tests.graphql.GraphqlTestCase.testCreateAndPublishGraphQLAPIUsingSchemaWithMalformedContext(GraphqlTestCase.java:226)
2023-07-09T20:07:49.5612426Z 
2023-07-09T20:07:49.5615194Z [ERROR] org.wso2.am.integration.tests.websocket.WebSocketAPITestCase.testCreateWebSocketAPIWithMalformedContext  Time elapsed: 0.154 s  <<< FAILURE!
2023-07-09T20:07:49.5615809Z java.lang.NullPointerException
2023-07-09T20:07:49.5616478Z 	at org.wso2.am.integration.tests.websocket.WebSocketAPITestCase.testCreateWebSocketAPIWithMalformedContext(WebSocketAPITestCase.java:498)
2023-07-09T20:07:49.5618937Z 
2023-07-09T20:07:49.5691707Z [ERROR] org.wso2.am.integration.tests.websocket.WebSocketAPITestCase.testCreateWebSocketAPIWithMalformedContext  Time elapsed: 0.151 s  <<< FAILURE!
2023-07-09T20:07:49.5692599Z java.lang.NullPointerException
2023-07-09T20:07:49.5693331Z 	at org.wso2.am.integration.tests.websocket.WebSocketAPITestCase.testCreateWebSocketAPIWithMalformedContext(WebSocketAPITestCase.java:498)
2023-07-09T20:07:49.5696010Z 
2023-07-09T20:07:49.5701486Z [ERROR] org.wso2.am.integration.tests.other.MandatoryPropertiesTestWithRestart.testCreateAnAPIThroughThePublisherRestWithoutMandatoryProperties  Time elapsed: 8.476 s  <<< FAILURE!
2023-07-09T20:07:49.5702466Z java.lang.AssertionError: APIs cannot be updated without setting values to mandatory API properties.
2023-07-09T20:07:49.5703484Z 	at org.wso2.am.integration.tests.other.MandatoryPropertiesTestWithRestart.testCreateAnAPIThroughThePublisherRestWithoutMandatoryProperties(MandatoryPropertiesTestWithRestart.java:110)
```

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/graphql/GraphqlTestCase.java [hunk 1]
```diff
@@ -189,6 +189,44 @@ public void createAndPublishGraphQLAPIUsingSchemaWithInterfaces() throws Excepti
...
+                        + "schemaWithInterface.graphql"), "UTF-8");
+        File file = getTempFileWithContent(schemaDefinitionWithInterface);
+        GraphQLValidationResponseDTO responseApiDto = restAPIPublisher.validateGraphqlSchemaDefinition(file);
+        GraphQLValidationResponseGraphQLInfoDTO graphQLInfo = responseApiDto.getGraphQLInfo();
+        String arrayToJson = new ObjectMapper().writeValueAsString(graphQLInfo.getOperations());
+        JSONArray operations = new JSONArray(arrayToJson);
+        HttpResponse response = null;
+
+        ArrayList<String> policies = new ArrayList<String>();
+        policies.add("Unlimited");
```
Comment (npamudika) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/graphql/GraphqlTestCase.java:207]: Shall we use APIMIntegrationConstants.API_TIER.UNLIMITED?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/graphql/GraphqlTestCase.java [hunk 2]
```diff
@@ -189,6 +189,44 @@ public void createAndPublishGraphQLAPIUsingSchemaWithInterfaces() throws Excepti
...
+
+        JSONObject additionalPropertiesObj = new JSONObject();
+        additionalPropertiesObj.put("name", "GraphQLAPIWithInvalidContext");
+        additionalPropertiesObj.put("context", "invalidContext{version}");
+        additionalPropertiesObj.put("version", API_VERSION_1_0_0);
```
Comment (npamudika) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/graphql/GraphqlTestCase.java:212]: Let's move the name, context, version words to constants.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/graphql/GraphqlTestCase.java [hunk 3]
```diff
@@ -189,6 +189,44 @@ public void createAndPublishGraphQLAPIUsingSchemaWithInterfaces() throws Excepti
...
+
+        JSONObject url = new JSONObject();
+        url.put("url", END_POINT_URL);
+        JSONObject endpointConfig = new JSONObject();
+        endpointConfig.put("endpoint_type", "http");
+        endpointConfig.put("sandbox_endpoints", url);
+        endpointConfig.put("production_endpoints", url);
+        additionalPropertiesObj.put("endpointConfig", endpointConfig);
+        additionalPropertiesObj.put("policies", policies);
+        additionalPropertiesObj.put("operations", operations);
```
Comment (npamudika) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/graphql/GraphqlTestCase.java:222]: Can move the property names to constants, as there are multiple usages in the class.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/publisher/APIM18CreateAnAPIThroughThePublisherRestAPITestCase.java [hunk 4]
```diff
@@ -149,6 +149,30 @@ public void testCreateAnAPIThroughThePublisherRest() throws Exception {
...
+        String apiTag = "tag18-4, tag18-5, tag18-6";
+        String apiName = "APIM18PublisherTestMalformed";
+
+        APIRequest apiCreationRequestBean;
+        apiCreationRequestBean = new APIRequest(apiName, apiContextTest, new URL(apiProductionEndPointUrl));
+
+        apiCreationRequestBean.setVersion(apiVersion);
+        apiCreationRequestBean.setDescription(apiDescription);
+        apiCreationRequestBean.setTags(apiTag);
+        apiCreationRequestBean.setTier("Gold");
```
Comment (npamudika) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/publisher/APIM18CreateAnAPIThroughThePublisherRestAPITestCase.java:169]: Shall we use APIMIntegrationConstants.API_TIER.GOLD?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/publisher/APIM18CreateAnAPIThroughThePublisherRestAPITestCase.java [hunk 5]
```diff
@@ -149,6 +149,30 @@ public void testCreateAnAPIThroughThePublisherRest() throws Exception {
...
+
+        HttpResponse response = restAPIPublisher.addAPIWithMalformedContext(apiCreationRequestBean);
+        Assert.assertNotNull(response, "Response cannot be null");
+        Assert.assertEquals(response.getResponseCode(), 400, "Response Code miss matched when creating the API");
```
Comment (npamudika) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/publisher/APIM18CreateAnAPIThroughThePublisherRestAPITestCase.java:173]: Let's use Response.Status.NOT_FOUND.getStatusCode() for 400

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/websocket/WebSocketAPITestCase.java [hunk 6]
```diff
@@ -467,6 +468,28 @@ public void testWebSocketAPIInvalidTokenInvocation() throws Exception {
...
+        //Create the api creation request object
+        apiRequest = new APIRequest(apiNameWithMalformedContext, apiContext, endpointUri, endpointUri);
+        apiRequest.setVersion(apiVersion);
+        apiRequest.setTiersCollection(APIMIntegrationConstants.API_TIER.ASYNC_UNLIMITED);
+        apiRequest.setProvider(provider);
+        apiRequest.setType("WS");
+        apiRequest.setApiTier(APIMIntegrationConstants.API_TIER.UNLIMITED);
+
+        HttpResponse response = restAPIPublisher.addAPIWithMalformedContext(apiRequest);
+        Assert.assertEquals(response.getResponseCode(), 400, "Response Code miss matched when creating the API");
```
Comment (npamudika) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/websocket/WebSocketAPITestCase.java:490]: Let's use Response.Status.NOT_FOUND.getStatusCode() for 400

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/graphql/GraphqlTestCase.java (hunk 1 above)
Comment (dulithsenanayake) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/graphql/GraphqlTestCase.java]: Fixed from a31b2d2b6166588d5fbdbd3331fdc42466a9070c

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/graphql/GraphqlTestCase.java (hunk 2 above)
Comment (dulithsenanayake) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/graphql/GraphqlTestCase.java]: Fixed from a31b2d2b6166588d5fbdbd3331fdc42466a9070c

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/graphql/GraphqlTestCase.java (hunk 3 above)
Comment (dulithsenanayake) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/graphql/GraphqlTestCase.java]: Fixed from a31b2d2b6166588d5fbdbd3331fdc42466a9070c

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/publisher/APIM18CreateAnAPIThroughThePublisherRestAPITestCase.java (hunk 4 above)
Comment (dulithsenanayake) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/publisher/APIM18CreateAnAPIThroughThePublisherRestAPITestCase.java]: Fixed from a31b2d2b6166588d5fbdbd3331fdc42466a9070c

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/publisher/APIM18CreateAnAPIThroughThePublisherRestAPITestCase.java (hunk 5 above)
Comment (dulithsenanayake) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/publisher/APIM18CreateAnAPIThroughThePublisherRestAPITestCase.java]: Fixed from a31b2d2b6166588d5fbdbd3331fdc42466a9070c

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/websocket/WebSocketAPITestCase.java (hunk 6 above)
Comment (dulithsenanayake) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/websocket/WebSocketAPITestCase.java]: Fixed from a31b2d2b6166588d5fbdbd3331fdc42466a9070c

**Review by npamudika — APPROVED** [review]

**Pull request merged**